homeassistant/components/ecobee/climate.py

```diff
@@ -564,26 +562,11 @@ def set_preset_mode(self, preset_mode: str) -> None:
         if self.preset_mode == PRESET_VACATION:
             self.data.ecobee.delete_vacation(self.thermostat_index, self.vacation)
 
-        if preset_mode == PRESET_AWAY:
+        if preset_mode == PRESET_AWAY_INDEFINITELY:
             self.data.ecobee.set_climate_hold(
                 self.thermostat_index, "away", "indefinite", self.hold_hours()
             )
-
-        elif preset_mode == PRESET_TEMPERATURE:
```

Comment:
Removed because there's a more direct way to set temperature.

Reply:
All these were you are removing needs be outlined much better in the breaking change sections and also they should be deprecated instead of just removed here and now.

Please update the PR description and add deprecation time by using repair issues.

Reply:
This is addressed by my comment below:
https://github.com/home-assistant/core/pull/108636#discussion_r1461394187

To summarize again here, #105745 introduced preset mode validation in the parent `Climate` component which made it impossible to pass any of these presets into this function, because they were never added to `self.preset_modes`. My changes do not break anything that wasn't already broken.

As far as I can tell from scouring quite a lot of Home Assistant + ecobee content online, no one has ever used these. Because of that, and because there are already other, documented ways of doing these things, I don't see any problem with just removing them.